heat/core/operations.py

```diff
@@ -11,6 +11,127 @@
 
 __all__ = []
 
+def __binary_bit_op(method, t1, t2):
```

Comment:
I am not 100% sure it is necessary to have this as a separate function in contrast to __binary_op(). It also seems that it is almost a verbatim copy of the function below. Why would we needs this except the type checking and casting purposes? It would be more meaningful from a SE perspective to factor out the common parts from __binary_op and __bitwise_binary_op

Reply:
You're right. It is mostly a copy of the function and I am reluctant because of that, too. However, I couldn't find a really satisfying solution at the moment.
What's different is the not desired type casting at the end of each of the if branches at the first half of the function. It still may be not be the best solution, but I'll try to factor each branch out for instance, scalar and mixed pairs to reduce some code redundancy. 
Also, the function calls at the end uses the overloaded operators of the torch.tensor class. I don't think I can do much there.